src/util/io_limiter.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+use std::option::Option;
+use std::io::{Result, Write};
+
+use rocksdb::RateLimiter;
+
+const REFILL_PERIOD: i64 = 100 * 1000;
+const FARENESS: i32 = 10;
+pub const DEFAULT_SNAP_MIN_BYTES_PER_TIME: u64 = 128 * 1024;
+pub const DEFAULT_SNAP_MAX_BYTES_PER_TIME: u64 = 2 * 1024 * 1024;
+pub const DEFAULT_SNAP_MAX_BYTES_PER_SEC: u64 = 20 * 1024 * 1024;
+
+pub struct IOLimiter {
+    inner: Option<RateLimiter>,
+    min_bytes_per_time: i64,
+    max_bytes_per_time: i64,
+}
+
+impl IOLimiter {
+    pub fn new(min_bytes_per_time: u64, max_bytes_per_time: u64, bytes_per_sec: u64) -> IOLimiter {
```

Comment:
do we really need min_bytes_per_time and max_bytes_per_time?

/cc @huachaohuang

Reply:
Yes we do, but I think we can set default values for these two args, the user does not really have to see these.

Reply:
Why not just use the `singleburst_bytes` instead of `min_bytes_per_time` and `max_bytes_per_time`?

Reply:
If we set a large `max_bytes_per_sec`, say 50MB/s，`singleburst_bytes` will be 5MB/s, it might affect other threads that are also doing disk write.

Reply:
But we can use one argument instead of `max_bytes_per_time` and `min_bytes_per_time`.